File: python_programming_basics/while_loops/e_03_vacation.py
money_needed = float(input())
owned_money = float(input())
days_spending = 0
days_saving = 0
total_days = 0

while owned_money < money_needed:
    action = input()
    action_money = float(input())
    total_days += 1

    if action == "spend":
        owned_money -= action_money
        if owned_money < 0:
            owned_money = 0

    elif action == "save":
        owned_money += action_money

if days_spending == 5:
    print(f"You can't save the money.\n{total_days}")

else:
    print(f"You saved the money for {total_days} days.")

File: JavaScript Basics/05. While Loop/e-03-Vacation.js
function vacationSavings(input) {
    let goal = Number(input[0]);
    let currentMoney = Number(input[1]);
    let index = 2;

    let spendDays = 0;
    let totalDays = 0;

    while (currentMoney < goal) {
        let action = input[index];
        index++;
        let amount = Number(input[index]);
        index++;
        totalDays++;

        if (action === "spend") {
            spendDays++;
            currentMoney -= amount;
            if (currentMoney < 0) currentMoney = 0;
            if (spendDays === 5) {
                console.log("You can't save the money.");
                console.log(totalDays);
                return;
            }
        } else if (action === "save") {
            spendDays = 0;
            currentMoney += amount;
        }
    }

    console.log(`You saved the money for ${totalDays} days.`);
}